### ISSUE TITLE
Error out when running in network mode `host`

### ISSUE THREAD

**Issue by harbu** [docker-library/cassandra#138]
From docker-compose.yml

```
cassandra:
        container_name: my_cassandra
        image: cassandra
        network_mode: host
        restart: on-failure
        ports:
            - "7000:7000"
            - "7001:7001"
            - "7199:7199"
            - "9042:9042"
            - "9160:9160"
        environment:
            CASSANDRA_LISTEN_ADDRESS: localhost
            CASSANDRA_BROADCAST_ADDRESS: localhost
            CASSANDRA_RPC_ADDRESS: localhost
        volumes:
            - ./data/cassandra:/var/lib/cassandra
```

Even with `CASSANDRA_LISTEN_ADDRESS` set as show above, I receive error

```
2018-03-06T13:42:55.247126641Z ERROR [main] 2018-03-06 13:42:55,246 CassandraDaemon.java:708 - Local host name unknown: java.net.UnknownHostException: linuxkit-025000000001: linuxkit-025000000001: Name or service not known
```

**Comment by yosifkit** [comment]
Your yaml file works here on a plain Linux host. My guess is something to do with "Docker for Mac" not resolving `localhost` properly? Try running with `127.0.0.1` for `CASSANDRA_BROADCAST_ADDRESS` (or maybe all of them).

**Comment by harbu** [comment]
I learned later that `network_mode: host` doesn't work with Docker for Mac thus I reverted back to bridged mode. The problem is that in "Docker for Mac" the images are run inside a separate Linux VM and appropriate ports are not exposed when using `network_mode: host`. This might be related to the problem I was having describing here as well.

For now, I'm closing this issue. Thank you for your assistance!
